lampod/src/ln/peer_manager.rs

```diff
@@ -177,13 +179,12 @@ impl LampoPeerManager {
                             })
                             .await;
                         }
-                        Err(e) => {
-                            log::error!(target: "lampo", "Error accepting connection: {}", e);
-                            return Err::<(), error::Error>(e.into());
-                        }
                     }
                 }
             });
+            if let Err(err) = result {
+                log::error!(target: "lampo", "{err}");
+            }
```

Comment:
This is a good option to log here but, but we should also return the error to the caller, so use the `?` on the `async_run!`

Reply:
Hey, we cannot use `?` to propagate the error to the caller as the compiler says
```
cannot use the `?` operator in a closure that returns `()`
```
so how do i proceed from here, 
should i implement a `mpsc` channel to send errors from the async block back to the main thread

Reply:
I need to run the code, I am not sure that the error that you point our is right because you are doing `let res = async_run`, so async_run is not a void.

Most of the time the compiler is wrong

Reply:
Please @VanshulB apply

```diff
diff --git a/lampod/src/ln/peer_manager.rs b/lampod/src/ln/peer_manager.rs
index a162e3b..279978a 100644
--- a/lampod/src/ln/peer_manager.rs
+++ b/lampod/src/ln/peer_manager.rs
@@ -161,11 +161,10 @@ impl LampoPeerManager {
                 loop {
                     let peer_manager = peer_manager.clone();

-                    match listener
-                        .accept()
-                        .await
-                        .map_err(|err| error::anyhow!("Error accepting connection: {}", err))?
-                    {
+                    let accept = listener.accept().await;
+                    let accept = accept
+                        .map_err(|err| error::anyhow!("Error accepting connection: {}", err))?;
+                    match accept {
                         (tcp_stream, _) => {
                             log::info!(target: "lampo", "Got new connection {}", tcp_stream.peer_addr().unwrap());
                             let _ = tokio::spawn(async move {
@@ -182,9 +181,10 @@ impl LampoPeerManager {
                     }
                 }
             });
-            if let Err(err) = result {
-                log::error!(target: "lampo", "{err}");
+            if let Err(err) = &result {
+                log::error!("error while try to listen on inbound connection: `{err}`");
             }
+            result
         });
```